src/simulation/hdf_simulation_store.jl

```diff
@@ -734,18 +767,55 @@ function _read_data_columns(
     return data, columns
 end
 
+function _read_data_columns(
+    store::HdfSimulationStore,
+    model_name::Symbol,
+    key::OptimizationContainerKey,
+    index::EmulationModelIndexType,
+)
+    if is_cached(store.cache, model_name, key, index)
+        # data = read_result(store.cache, model_name, key, index)
+        #columns = _get_em_dataset(store, model_name, key).column_dataset[:]
+    else
+        data, columns = _read_result(store, model_name, key, index)
+    end
+
+    return data, columns
+end
+
 function _read_length(::Type{OptimizerStats}, store::HdfSimulationStore)
     dataset = _get_dataset(OptimizerStats, store)
     return HDF5.read(HDF5.attributes(dataset), "columns")
 end
 
-function _write_dataset!(dataset, array::Array{Float64, 2}, row_range::UnitRange{Int64})
+function write_dataset!(
```

Comment:
Previously, this function started with an underscore because it was private to this file. It was not intended to be used externally because it writes to the HDF5 file. The external interface needs to account for the cache. Has any of that changed?

Reply:
I changed the name to use it in `set_value!()` in the dataset interface

Reply:
@daniel-thom fixed via e75c5af